packages/workspace/src/core/affected-project-graph/locators/tsconfig-json-changes.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  isWholeFileChange,
+  normalizedProjectRoot,
+  WholeFileChange
+} from '../../file-utils';
+import { DiffType, isJsonChange, JsonChange } from '../../../utils/json-diff';
+import { TouchedProjectLocator } from '../affected-project-graph-models';
+import { isWorkspaceProject } from '../../project-graph';
+import { join, relative } from 'path';
+import { appRootPath } from '../../../utils/app-root';
+
+export const getTouchedProjectsFromTsConfig: TouchedProjectLocator<
```

Comment:
Could you split the function into smaller functions with long names, types, and comments?

The code is very dense. I think it works correctly, but, but it's hard to tell.